src/main/java/net/minecraftforge/client/event/RecipesUpdatedEvent.java

```diff
@@ -0,0 +1,30 @@
+package net.minecraftforge.client.event;
+
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.item.crafting.RecipeManager;
+import net.minecraftforge.api.distmarker.Dist;
+import net.minecraftforge.eventbus.api.Event;
+
+import java.util.Collection;
+import java.util.Collections;
+
+/**
+ * Fired when {@link RecipeManager} has its recipes updated.
+ *
+ * This happens on {@link Dist#CLIENT} when recipes are synced from the server to the client (just after a client has connected),
+ * and on both {@link Dist#DEDICATED_SERVER} and {@link Dist#CLIENT} when resource packs are loaded or reloaded.
+ */
+public class RecipesUpdatedEvent extends Event
+{
+    private final Collection<IRecipe> recipes;
+
+    public RecipesUpdatedEvent(Collection<IRecipe> recipes)
+    {
+        this.recipes = Collections.unmodifiableCollection(recipes);
+    }
+
+    public Collection<IRecipe> getRecipes()
```

Comment:
maybe some docs on what exactly this holds? All the recipes or the ones that changed/reloaded?

Reply:
It's always the same as `RecipeManager#getRecipes` so I'm actually thinking of removing this from the event.
Thoughts?

Reply:
As long as that fact is documented in `RecipesUpdatedEvent`, that should be fine.